src/community/oseo/oseo-stac/src/test/java/org/geoserver/ogcapi/v1/stac/ItemsTest.java

```diff
@@ -87,6 +87,20 @@ public void testSentinelItemsJSON() throws Exception {
         assertEquals("2017-02-26T10:30:00.031+00:00", s2Range.read("properties.end_datetime"));
     }
 
+    @Test
+    public void testSentinelItemsJSONWorkspace() throws Exception {
+        // Mismatched workspace
+        getAsMockHttpServletResponse("cite/ogc/stac/v1/collections/SENTINEL2/items?limit=50", 404);
+        // Matched workspace
+        DocumentContext docMatchingWorkspace =
+                getAsJSONPath("sf/ogc/stac/v1/collections/SENTINEL2/items?limit=50", 200);
+        assertEquals(Integer.valueOf(19), docMatchingWorkspace.read("numberReturned"));
+        // No match, but has null in workspace array
+        DocumentContext docNullWorkspace =
+                getAsJSONPath("sf/ogc/stac/v1/collections/SAS1/items?limit=50", 200);
+        assertEquals(Integer.valueOf(19), docMatchingWorkspace.read("numberReturned"));
```

Comment:
Uh oh, not a match for the initial requirements, quoting:

* A collection with null value shows up in all workspaces and in global services too
* A collection shows up in a workspace if one of the names in the array matches the workspace name
* A collection with a "null" entry in the array shows up in global services

Having null in the array means the collection should show in the global services, not in all workspace services.